src/utils/generator.js

```diff
@@ -597,7 +597,7 @@ export class Generator {
     } else if (result instanceof JoinedNormalAndAbruptCompletions) {
       let selector = c =>
         c instanceof ThrowCompletion && c.value !== realm.intrinsics.__bottomValue && !(c.value instanceof EmptyValue);
-      output.emitConditionalThrow(Join.joinValuesOfSelectedCompletions(selector, result));
+      output.emitConditionalThrow(Join.joinValuesOfSelectedCompletions(selector, result, true));
```

Comment:
It's hard to tell from the callsite what the `, true` argument means. Maybe make it named?

Why don't you want to `keepInfeasiblePaths` here?